### PR TITLE
Move Sampler builder to top level

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/ParentBasedSampler.java [hunk 1]
```diff
@@ -19,7 +19,7 @@
 /**
  * A Sampler that uses the sampled flag of the parent Span, if present. If the span has no parent,
  * this Sampler will use the "root" sampler that it is built with. See documentation on the {@link
- * Builder} methods for the details on the various configurable options.
+ * ParentBasedSamplerBuilder} methods for the details on the various configurable options.
  */
 @Immutable
 public class ParentBasedSampler implements Sampler {
```
Comment (jkwatson) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/ParentBasedSampler.java:25]: can we make this class non-public now? That would be an improvement, if we can.

**Review by jkwatson — APPROVED** [review]
Thanks!

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/ParentBasedSampler.java (hunk 1 above)
Comment (bogdandrutu) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/ParentBasedSampler.java:25]: Also final :). Cannot be non-public if the "builder" method is on this class.

Comment (jkwatson) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/ParentBasedSampler.java:25]: yeah, I realized that and forgot to remove the comment.

Comment (anuraaga) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/ParentBasedSampler.java:25]: Nice, thanks!

**Pull request merged**